ios/src/TGFontConverter.mm

```diff
@@ -0,0 +1,154 @@
+//
+//  TGFontConverter.m
+//  tangram
+//
+//  Created by Karim Naaji on 11/01/16.
+//
+//
+
+#import "TGFontConverter.h"
+
+#include <cstdio>
+
+struct FontHeader {
+    int32_t fVersion;
+    uint16_t fNumTables;
+    uint16_t fSearchRange;
+    uint16_t fEntrySelector;
+    uint16_t fRangeShift;
+};
+
+typedef struct FontHeader FontHeader;
+
+struct TableEntry {
+    uint32_t fTag;
+    uint32_t fCheckSum;
+    uint32_t fOffset;
+    uint32_t fLength;
+};
+
+typedef struct TableEntry TableEntry;
+
+static uint32_t CalcTableCheckSum(const uint32_t* table, uint32_t numberOfBytesInTable)
+{
+    uint32_t sum = 0;
+    uint32_t nLongs = (numberOfBytesInTable + 3) / 4;
+    while (nLongs-- > 0) {
+       sum += CFSwapInt32HostToBig(*table++);
+    }
+    return sum;
+}
+
+static uint32_t CalcTableDataRefCheckSum(CFDataRef dataRef)
+{
+    const uint32_t* dataBuff = (const uint32_t *)CFDataGetBytePtr(dataRef);
+    uint32_t dataLength = (uint32_t)CFDataGetLength(dataRef);
+    return CalcTableCheckSum(dataBuff, dataLength);
+}
+
+@implementation TGFontConverter
+
+//Reference:
+//http://skia.googlecode.com/svn-history/r1473/trunk/src/ports/SkFontHost_mac_coretext.cpp
```

Comment:
This, I think: https://skia.googlesource.com/skia/+/master/src/ports/SkFontHost_mac.cpp